ncoreparser/client.py

```diff
@@ -79,7 +79,7 @@ def search(self, pattern, type=SearchParamType.ALL_OWN, where=SearchParamWhere.N
             except Exception as e:
                 raise NcoreConnectionError(f"Error while searhing torrents. {e}") from e
             new_torrents = [Torrent(**params) for params in self._page_parser.get_items(request.text)]
-            if number is None or len(new_torrents) == 0:
+            if len(new_torrents) == 0:
```

Comment:
By default the number=None, so if you want to get all torrents you don't need to specify the number. I think this wont end up in a infinite loop, because the exit point is when the newly fetched torrents length is 0, so no new torrents. If somehow the request fails, that exits too. If no torrents at all, that exits too.

Reply:
I have tested it with this code snipped:
```python
torrents = client.search(pattern="")
for torrent in torrents:
    pretty_print(torrent)
```
And it occurred infinite loop. So the check is needed here.
Please try it also.
I recommend to modify the manual test and add this section:
```python
print_category("Test number=None")
torrents = client.search(pattern="")
for torrent in torrents:
    pretty_print(torrent)
``` 
According to my solution if the number is None it will return one one page size what is default 25.

Reply:
Okay, I'll check this out later. Thanks.

Reply:
What about my latest change? 

The infinit loop's was because of the missing pattern was matching for every torrent (i guess) and it was fetching everything or something like that, because the torrents array was bigger by 100 new items on every iterations.

Now, if no pattern is defined no need to do anything. This way no problem not defining the number, because this solution will return all of the matching torrent for that specific pattern.

Later probably I'll make some change to this function to be able to define the page that I want to retrieve the torrents from, but for now this should be good.

Reply:
What about the pattern is a short letter like "a"? It also wil occurre long running time.

Note: the async version of the client has not changed. 

I recommend to add -1 as option for number argument. In this case it will load all findings. But I would keep the None as default value.

Reply:
Looks good to me. Thanks the idea.